decentraland/unity-explorer
# fix: trigger scene emotes for local development

### WHY

`triggerSceneEmote` calls from the SDK were not working for local scene development (LSD), the reason being the code path for triggering scene emotes was highly coupled with AssetBundles and did not consider LSD at all. The code path for `triggerSceneEmote` is completly different than everything related to load raw gltf for LSD so it was overlooked.

### WHAT

Implemented a branch to trigger scene emotes both for LSD and AssetBundle normal flow. Implemented a bit of abstraction to introduce as less branching as possible and streamline the usage of asset data to trigger emotes. Fixed animation clips loading for raw gltf loading in LSD (not needed for AssetBundle's flow)

Fixes https://github.com/decentraland/unity-explorer/issues/2249

------------------------------------

### TEST STEPS

1. Download [this test scene](https://github.com/user-attachments/files/17370014/Scene.Emotes.zip)
2. Unzip it somewhere and enter its root folder with a console/terminal
3. run `npm i`
4. run `npm run start -- --explorer-alpha`
5. Confirm that with the latest published E@ the scene emotes don't play by clicking the white boxes (2 should trigger Scene Emotes, one triggers normal Avatar Emotes)
6. Close the Explorer but leave the scene running in the terminal
7. Download the build from this PR
8. Connect the PR build to the locally running scene, following [the steps in the Wiki](https://github.com/decentraland/unity-explorer/wiki/How-to-connect-to-a-local-scene-(Unity-Editor---Custom-Build---Latest-Released-Build)#connecting-a-custom-build-to-the-scene) (remember to put the correct scene position parameter, in this case: --position 80,-1)
9. Confirm that now you actually can trigger the 2 Scene Emotes, Craft Snowball and Throw Snowball
